lib/src/search_page.dart

```diff
@@ -0,0 +1,141 @@
+  void initState() {
+    super.initState();
+    filteredCountries = allCountries; // Initialize with all countries
+  }
+
+  void updateFilteredCountries(List<Map<String, String>> countries) {
+    setState(() {
+      filteredCountries = countries;
+    });
+  }
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Select a Country"),
+      ),
+      body: Column(
+        children: [
+          CountrySearchBar(
+            allCountries: allCountries,
+            onSearch: updateFilteredCountries,
+          ),
+          Expanded(
+            child: ListView.builder(
+              itemCount: filteredCountries.length,
+              itemBuilder: (context, index) {
+                final country = filteredCountries[index];
+                return GestureDetector(
+                  onTap: () {
+                    // Pass the selected country back to the previous page
+                    Navigator.pop(context, country);
+                  },
+                  child: Container(
+                    padding: EdgeInsets.symmetric(vertical: 10, horizontal: 16),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                      children: [
+                        Row(
+                          children: [
+                            Text(
+                              '${Countries.getFlagEmoji(country['code']!)}',
+                              style: TextStyle(fontSize: 24),
+                            ),
+                            SizedBox(width: 8), // Add some spacing between flag and dial_code
+                            Text(
+                              '${country['name']}',
+                              style: TextStyle(fontSize: 16),
+                              overflow: TextOverflow.ellipsis,
+                            ),
+                          ],
+                        ),
+                        Text(
+                          country['dial_code']!,
+                          style: TextStyle(fontSize: 16),
+                          overflow: TextOverflow.ellipsis,
+                        ),
+                      ],
+                    ),
+                  ),
+                );
+
+              },
+            ),
+          ),
+        ],
+      ),
+    );
+
+  }
+}
+
+class CountrySearchBar extends StatefulWidget {
+  final List<Map<String, String>> allCountries;
+  final Function(List<Map<String, String>>) onSearch;
+
+  CountrySearchBar({required this.allCountries, required this.onSearch});
+
+  @override
+  _CountrySearchBarState createState() => _CountrySearchBarState();
+}
+
+class _CountrySearchBarState extends State<CountrySearchBar> {
+  TextEditingController _searchController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    _searchController.addListener(_onSearchTextChanged);
```

Comment:
dispose of the controller before widget dispose as to remove the listener

Reply:
done